engine/src/main/resources/org/camunda/bpm/engine/impl/mapping/entity/HistoricProcessInstance.xml

```diff
@@ -409,7 +409,7 @@
         <bind name="JOIN_TYPE" value="'left join'" />
       </if>
 
-      <if test="query != null &amp;&amp; (query.withIncidents || query.withRootIncidents || query.incidentStatus != null || query.incidentMessage != null || query.incidentMessageLike != null || query.incidentType != null)">
+      <if test="query != null &amp;&amp; (query.withIncidents || query.withRootIncidents || query.incidentStatus != null || query.incidentMessage != null || query.incidentMessageLike != null || query.incidentType != null || (incidentIds != null &amp;&amp; incidentIds.length > 0))">
```

Comment:
❓ Do we need to extend to docs that incidentIds are excluded for Or queries here: https://docs.camunda.org/manual/develop/user-guide/process-engine/process-engine-api/#or-queries

Reply:
Good point! I added a note to [Start date filter for modify operation - Documentation](https://github.com/camunda/camunda-bpm-platform/issues/4632)

Reply:
❓ Are they excluded? 🤔

Reply:
Maybe we can just add a test to verify this.